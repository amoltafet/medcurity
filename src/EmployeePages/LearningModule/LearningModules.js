import React from "react";

/**
 * Returns a essentially blank page. Serves to handle the postSlug.
 * @returns 
 */
function LearningModules() {
  return (
    <div >
      <div >
        <h1 className="text-center mt-5">Custom Learning Module page?</h1>
      </div>
    </div>
  );
}

export default LearningModules;