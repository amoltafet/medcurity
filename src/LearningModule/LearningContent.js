import React from 'react';
import './LearningContent.css';

/**
 * 
 * @param {Array<str>} answers for the question
 * @param {str} question The question string
 * @param {int} number The question number
 */
function LearningContent(props) {

    return (
        <>
            <h1>{props.title}</h1>
            <div>{props.paragraph}</div>
        </>
    );
}

export default LearningContent