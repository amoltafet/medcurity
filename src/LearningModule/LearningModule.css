.learningModuleBg {
    background-image: url("learningModuleBg.png");
    height: 90vh;
    width: 100%;
    margin-top: 1%;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    position: fixed;
}

.moduleName {
    color: #1e5b88;
    margin-top: -2%;
    font-family: Lato,sans-serif;
    font-size: 250%;
}

.moduleImage{
    margin-top: 10%;
}

.moduleSubtitle {
    color: #127ebf;
    font-family: Lato,sans-serif;
    font-size: 150%;
}

.moduleDescription {
    color: black;
    margin-left: 7%;
    margin-right: 6%;
    font-family: Lato,sans-serif;
    font-size: 100%;
}

.goToQuizBttn {
    margin-right: 40%;
    margin-left: 40%;
    width: 20%;
    margin-top: 2%;
    background-color: #127ebf;
    font-family: Lato,sans-serif;
}

.uvs-left {
    border-radius: 10px 10px 10px 10px;
    box-shadow:  0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}
    
    
.uvs-right {
    border-radius: 10px 10px 10px 10px;
    
}