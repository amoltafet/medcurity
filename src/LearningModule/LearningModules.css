.uvs-left {
    border-radius: 10px 10px 10px 10px;
}
      
.uvs-right {
    border-radius: 10px 10px 10px 10px;
    
}