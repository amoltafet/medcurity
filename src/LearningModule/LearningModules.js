import React from "react";
import { useState, useEffect } from "react";
import axios from 'axios';
import { Link } from "react-router-dom"

/**
 * Returns a essentially blank page. Serves to handle the postSlug.
 * @returns 
 */
function LearningModules() {
  const [content, setContent] = useState([])
  const string = 'LearningModules'

  useEffect(() => {
    axios.get('http://localhost:3002/api/getData', { params: { table: string } }).then((response) => {
          setContent(Object.values(response.data))
      });
  }, [])

  return (
    <div className="home">
      <div class="container">
        <h1 className="text-center mt-5">Custom Learning Module page???</h1>
        {content.map((val) => {
          return (<h2>
            Description: {val.Description}
          </h2>);
        })}
      </div>
    </div>
  );
}

export default LearningModules;