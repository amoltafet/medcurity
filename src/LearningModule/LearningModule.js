import { Button } from 'react-bootstrap'
import React from 'react';
import './LearningModule.css';
import 'bootstrap/dist/css/bootstrap.min.css'
import { useEffect, useState } from "react";
import { useParams } from "react-router";
import axios from 'axios';

/**
* Creates and displays the learning page for each test category. 
* @return { LearningModule }
*/
const  LearningModule = () => {
    let { slug } = useParams();
    const [content, setContent] = useState([])

    useEffect(() => {
      // Fetch post using the postSlug
    }, [slug]);

    useEffect(() => {
        axios.get('http://localhost:3002/api/getModuleInfo', { params: { id: slug } }).then((response) => {
              setContent(Object.values(response.data))
          });
    }, [])


    const LearningModuleContent = content.map((module) => {
        return ([
            <h1 className="text-center moduleName">
              Learning Modules: {module.Title} Module
            </h1>,
            <div className="d-flex justify-content-center">
              <img src={require(`../assets/` + module.Banner)} className="img-fluid rounded mx-auto d-block moduleImage uvs-left uvs-right" alt={module.Title} />
            </div>,
            <h6 className="text-center mt-2 moduleSubtitle">
              {module.Subtitle}
            </h6>,
            <h4 className="text-center mt-3 moduleDescription">
              {module.Description}
            </h4>
        ]);
    })

    return (
        <>
         <div className="learningModuleBg img-fluid ">
        {LearningModuleContent}
        <div className="d-grid gap-2 ">
            <Button variant="primary" className="goToQuizBttn uvs-left uvs-right" href={'/quiz/' + slug}>
                Go to Quiz
            </Button>
          </div>
        </div>
        </>
    );
}
export default  LearningModule;
