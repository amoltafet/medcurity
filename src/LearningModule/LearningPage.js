import {Button,  Container} from 'react-bootstrap'
import React from 'react';
import './LearningPage.css';
import 'bootstrap/dist/css/bootstrap.min.css'
import MenuBar from '../MenuBar';
import LearningContent from './LearningContent.js';


/**
* Creates and displays the learning page for each test category. 
* @return {LearningPage}
*/
const LearningPage = () => {
    const myanswers = ["42", "Life is the meaning of life", "13"];
    const question = "What is the meaning of life?";
    const number = 1;

    return (
        <>
        <MenuBar></MenuBar>
        <Container className="learningPageContainer">
            <LearningContent title="Privacy" paragraph = "This is some text." />
        </Container>
        <div className="d-grid gap-2">
            <Button variant="primary" href="/quiz">
                "Go to Quiz"
            </Button>
        </div>
        </>
    );
}
export default LearningPage;