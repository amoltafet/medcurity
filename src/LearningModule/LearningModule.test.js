import renderer from 'react-test-renderer';
import LearningModule from './LearningModule';
import React from 'react';

test('renders LearningModule', () => {
  const tree = renderer.create(<LearningModule />).toJSON();
  expect(tree).toMatchSnapshot();
});