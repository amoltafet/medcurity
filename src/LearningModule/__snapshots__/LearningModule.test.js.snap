// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LearningModule 1`] = `
<div
  className="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-15lsh1k-MuiGrid2-root"
>
  <div
    className="learningModuleBg img-fluid "
  >
    <div
      className="justify-content-center row row-cols-lg-10 row-cols-md-10 row-cols-10"
    >
      <div
        className="col-lg-3 col-md-3 col-8"
      >
        <a
          className="MuiButtonBase-root MuiButton-root MuiButton-primary MuiButton-primaryPrimary MuiButton-sizeMedium MuiButton-primarySizeMedium MuiButton-root MuiButton-primary MuiButton-primaryPrimary MuiButton-sizeMedium MuiButton-primarySizeMedium css-jt6ki8-MuiButtonBase-root-MuiButton-root"
          href="/dash"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          Back to Dashboard
        </a>
      </div>
      <div
        className="col-lg-3 col-md-3 col-8"
      >
        <a
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-xsqe0f-MuiButtonBase-root-MuiButton-root"
          href="/quiz/undefined"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          Start 
          
           Quiz
        </a>
      </div>
    </div>
  </div>
</div>
`;
